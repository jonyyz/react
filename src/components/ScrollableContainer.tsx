import * as React from "react";

import {
  makeStyles,
  createStyles,
} from "@material-ui/styles";

const useStyles = makeStyles(createStyles({
  scrollable: {
    overflowY: "auto",
    overflowX: "auto",
    "-webkit-overflow-scrolling": "touch",
  }
}));

const ScrollableContainer: React.FunctionComponent = ({ children }) => {
  const classes = useStyles({});

  return (
    <div className={classes.scrollable}>{children}</div>
  );
}

export default ScrollableContainer;
