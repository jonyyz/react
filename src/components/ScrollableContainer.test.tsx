import * as React from "react";
import { shallow } from "enzyme";

import ScrollableContainer from "./ScrollableContainer";

describe("ScrollableContainer", () => {
  test("should render with children", () => {
    const wrapper = shallow(
      <ScrollableContainer>
        <div id="test">test</div>
      </ScrollableContainer>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
