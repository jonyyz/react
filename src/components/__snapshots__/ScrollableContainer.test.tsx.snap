// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollableContainer should render with children 1`] = `
<div
  className="makeStyles-scrollable-1"
>
  <div
    id="test"
  >
    test
  </div>
</div>
`;
